blog/views.py

```diff
@@ -20,3 +22,36 @@ def post_detail(request, pk):
     """Display a blog post based on the post's primary key."""
     post = get_object_or_404(Post, pk=pk)
     return render(request, 'blog/post_detail.html', {'post': post})
+
+
+@login_required
+def post_new(request):
+    """Create a new post."""
+    if request.method == 'POST':
+        form = PostForm(request.POST)
+        if form.is_valid():
+            post = form.save(commit=False)
+            post.author = request.user
+            post.published_date = timezone.now()
+            post.save()
+            return redirect('post_detail', pk=post.pk)
+    else:
+        form = PostForm
```

Comment:
Did you forget to initialize the object here?

Reply:
Yes

Reply:
marking as resolved